main.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/joho/godotenv"
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+	"github.com/slack-go/slack/socketmode"
+)
+
+func main() {
+	// Load env
+	godotenv.Load(".env")
+	token := os.Getenv("SLACK_AUTH_TOKEN")
+	appToken := os.Getenv("SLACK_APP_TOKEN")
+
+	// Set debug to true for dev
+	client := slack.New(token, slack.OptionDebug(true), slack.OptionAppLevelToken(appToken))
+	// Enable Socket Mode
+	socketClient := socketmode.New(
+		client,
+		socketmode.OptionDebug(true),
+		// Custom logger
+		socketmode.OptionLog(log.New(os.Stdout, "socketmode: ", log.Lshortfile|log.LstdFlags)),
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func(ctx context.Context, client *slack.Client, socketClient *socketmode.Client) {
+		for {
+			select {
+			case <-ctx.Done():
+				log.Println("Shutting down socketmode listener")
+				return
+			case event := <-socketClient.Events:
+				switch event.Type {
+				// EventAPI
+				case socketmode.EventTypeEventsAPI:
+					eventsAPIEvent, ok := event.Data.(slackevents.EventsAPIEvent)
+					if !ok {
+						log.Printf("Could not type cast the event to the EventsAPIEvent: %v\n", event)
+						continue
+					}
+					socketClient.Ack(*event.Request)
+					// log.Println(eventsAPIEvent)
+					err := handleEventMessage(eventsAPIEvent, client)
+					if err != nil {
+						//TODO: actual err handeling
```

Comment:
Would we finish todo in this patch ? 😆

Reply:
We haven't decide how to handle the error for this project.  Such as printing the log directly in console? or exporting logs to StackDriver?

Anyways, I'll unify all the error handling in bot to prevent bot stopped while error occured.